src/averageAges.js

```diff
@@ -37,7 +44,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const peopleWomenIsMotherAge = people.map(persone => ({
+    ...persone,
+    isMother: people.some(one => one.mother === persone.name),
+    age: persone.died - persone.born,
+  })).filter(persone => persone.sex === 'f');
+
+  const peopleMother
+  = peopleWomenIsMotherAge.filter(persone => persone.isMother);
+
+  return !withChildren ? (peopleWomenIsMotherAge.reduce((
+    a, { age }) => a + age, 0)) / peopleWomenIsMotherAge.length
+    : (peopleMother.reduce((
+      a, { age }) => a + age, 0)) / peopleMother.length;
```

Comment:
you have used this logic for calculations average age several times - lets create separate function and use it in all places